lib/socketpair.c

```diff
@@ -239,6 +239,7 @@ int Curl_socketpair(int domain, int type, int protocol,
   return -1;
 }
 #else
+#ifndef _WIN32
```

Comment:
Shouldn't this code block be dependent on `defined(HAVE_SOCKETPAIR) && !defined(CURL_DISABLE_SOCKETPAIR)` instead?

I think `HAVE_SOCKETPAIR` must be true as `socketpair` is used here, and `Curl_socketpair` must only be defined if `CURL_DISABLE_SOCKETPAIR` is not defined.

Reply:
You're right! I checked out the code elsewhere, it seemed pointless to define this function with `CURL_DISABLE_SOCKETPAIR` specified.